src/deluge/util/container/enum_to_string_map.hpp

```diff
@@ -0,0 +1,35 @@
+#include "definitions.h"
+#include "io/debug/log.h"
+#include "mem_functions.h"
+#include <array>
+#include <cstddef>
+#include <cstring>
+#include <utility>
+template <class Enum, std::size_t N>
+class EnumStringMap {
+	const char* stringList[N];
+
+public:
+	constexpr EnumStringMap(const std::array<std::pair<Enum, const char*>, N>& init) : stringList() {
+		for (std::size_t i = 0; i < N; ++i) {
+			const auto& p = init[i];
+			stringList[static_cast<std::underlying_type_t<Enum>>(p.first)] = p.second;
+		}
+	}
+
+	constexpr const char* operator()(Enum a) { return stringList[static_cast<std::underlying_type_t<Enum>>(a)]; }
+
+	constexpr Enum operator()(const char* str) {
+		for (int i = 0; i < N; i++) {
+			if (!strcmp(str, stringList[i])) {
+				return static_cast<Enum>(i);
+			}
+		}
+#if ALPHA_OR_BETA_VERSION
+		char popup[25];
+		sprintf(popup, "no match for:%s", str);
+		freezeWithError(popup);
```

Comment:
Does this work on 7seg?

Reply:
It does not, I'll just log it instead. It's probably better not to freeze on any unrecognized tag later anyway